Assets/LeapMotion/Core/Scripts/FollowHandPoint.cs

```diff
@@ -0,0 +1,248 @@
+using Leap.Unity.Query;
+using UnityEngine;
+
+namespace Leap.Unity.Attachments {
+  
+  [ExecuteInEditMode]
+  public class FollowHandPoint : MonoBehaviour, IStream<Pose> {
+
+    #region Inspector
+
+    [SerializeField]
+    private LeapProvider _provider;
+    public LeapProvider provider {
+      get {
+        if (_provider == null) { _provider = Hands.Provider; }
+        return _provider;
+      }
+    }
+
+    public Chirality whichHand;
+    
+    public AttachmentPointFlags attachmentPoint = AttachmentPointFlags.Palm;
+
+    public enum FollowMode { Update, FixedUpdate }
+    [SerializeField, OnEditorChange("followMode")]
+    private FollowMode _followMode = FollowMode.Update;
+    public FollowMode followMode {
+      get {
+        return _followMode;
+      }
+      set {
+        if (value != _followMode) {
+          if (Application.isPlaying) {
+            unsubscribeFrameCallback();
+          }
+
+          _followMode = value;
+
+          if (Application.isPlaying) {
+            subscribeFrameCallback();
+          }
+        }
+      }
+    }
+
+    [SerializeField]
+    [Disable]
+    private bool _isHandTracked = false;
+    public bool isHandTracked { get { return _isHandTracked; } }
+
+    [Header("Advanced")]
+
+    public bool showAdvancedSettings = false;
+
+    public bool usePalmRotationForWrist = false;
+
+    [Header("Pose Stream")]
+
+    [Tooltip("Follow Hand Point implements IStream<Pose>; It will stream data as long as "
+          + "the component is enabled, the hand is tracked, and this option is enabled.")]
+    [QuickButton("Send Stream Now", "sendStreamNow",
+      tooltip: "Open, send pose, and close the stream immediately. Edit-time only.")]
+    public bool doPoseStream = true;
+
+    [DisableIf("doPoseStream", isEqualTo: false)]
+    public bool usePoseStreamOffset = false;
+
+    [DisableIfAny("usePoseStreamOffset", "doPoseStream", areEqualTo: false)]
+    public Transform poseStreamOffsetSource = null;
+
+    [Disable]
+    public Pose poseStreamOffset = Pose.identity;
+
+    private bool _isStreamOpen = false;
+
+    #endregion
+
+    #region Unity Events
+
+    private void OnValidate() {
+      if (!usePoseStreamOffset) {
+        poseStreamOffset = Pose.identity;
+      }
+      else if (poseStreamOffsetSource != null) {
+        poseStreamOffset = transform.ToWorldPose().inverse
+                           * poseStreamOffsetSource.ToWorldPose();
+      }
+    }
+
+    void OnEnable() {
+      unsubscribeFrameCallback();
+      subscribeFrameCallback();
+    }
+
+    void OnDisable() {
+      if (_isStreamOpen) {
+        OnClose();
+        _isStreamOpen = false;
+      }
+
+      unsubscribeFrameCallback();
+    }
+
+    private void Update() {
+      if (!Application.isPlaying) {
+        moveToAttachmentPointNow();
+      }
+    }
+
+    #endregion
+
+    #region On Frame Event
+
+    private void onUpdateFrame(Frame frame) {
+      if (frame == null) Debug.Log("Frame null");
```

Comment:
Maybe a more descriptive error message?  Using an error path instead of a log path would also be great.

Reply:
:+1: